src/operator/tensor/elemwise_binary_op_basic.cc

```diff
@@ -64,12 +62,16 @@ NNVM_REGISTER_OP(_backward_mul)
   [](const NodeAttrs& attrs){
     return std::vector<std::pair<int, int> >{{0, 1}};
   })
-.set_attr<FCompute>("FCompute<cpu>", BinaryBackwardUseIn<cpu, mshadow_op::right,
-                                                              mshadow_op::left>);
+.set_attr<nnvm::FInferStorageType>("FInferStorageType", ElemwiseStorageType<3, 2>)
+.set_attr<FCompute>("FCompute<cpu>", BinaryOp::BinaryBackwardUseIn<cpu, mshadow_op::right,
+    mshadow_op::left>)
+.set_attr<FComputeEx>("FComputeEx<cpu>", BinaryOp::BinaryBackwardUseInEx<cpu, mshadow_op::right,
+    mshadow_op::left>);
+;
 
-MXNET_OPERATOR_REGISTER_BINARY(_div)
-.add_alias("_Div")
-.set_attr<FCompute>("FCompute<cpu>", BinaryCompute<cpu, mshadow::op::div>)
+// For divide, we will always auto-convert to dense since the sparse 0's will generate nans
+MXNET_OPERATOR_REGISTER_BINARY_LAUNCH_CPU_DR(elemwise_div, mshadow::op::div)
```

Comment:
Recall our conversation: 
for csr divide dns, we could pass an argument hint which indicates the user is confident that dns doesn't include any zeros, thus we only need to look at the non-zero entries in csr and corresponding entries in dns, which is fast. If the user doesn't give the hint, we have to look at all entries in dns. 
Does it make sense?